services/product-service/src/functions/getProductsById/handler.ts

```diff
@@ -1,25 +1,32 @@
-import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
-import { middyfy } from "@libs/lambda";
 import { APIGatewayProxyResult } from "aws-lambda/trigger/api-gateway-proxy";
-import products from "../../products.json";
+import { middyfy } from "@libs/lambda";
+import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { IDatabaseService } from "src/db-service";
+import { notFoundResponse, okResponse } from "src/responses";
+import { databaseService } from "src/dependencies/dependencies";
+
+const getProductsById =
+  (
+    databaseService: IDatabaseService
+  ): ValidatedEventAPIGatewayProxyEvent<APIGatewayProxyResult> =>
+  async (event) => {
+    console.log(event);
+
+    const productId = event?.pathParameters?.id;
```

Comment:
db error 500 is not handled (I also probably missed it). You could put it from this line in a try-catch block like in createProduct. Additionally, you could add 500 responses in swagger.